api/service/store.go

```diff
@@ -35,23 +37,41 @@ func addEndpoint(apiContext *types.APIContext, schema *types.Schema, data map[st
 		return data, nil
 	}
 
+	spaceID, _ := data["spaceId"].(string)
+	space := ""
+	parts := strings.SplitN(spaceID, "-", 2)
+	if len(parts) == 2 {
+		space = parts[1]
+	}
+
 	expose := false
+	protocol := ""
 	for _, ports := range convert.ToMapSlice(data[client.ServiceFieldPortBindings]) {
 		proto, _ := ports[client.PortBindingFieldProtocol].(string)
-		if proto != "http" {
+		if proto != "http" && proto != "https" {
 			continue
 		}
 		expose = true
+		protocol = proto
 		break
 	}
 
 	if expose {
 		data["endpoints"] = []interface{}{
 			map[string]interface{}{
-				"url": fmt.Sprintf("http://%s.%s.%s", name, stackName, domain),
+				"url": fmt.Sprintf("%s://%s.%s", protocol, HashIfNeed(name, stackName, space), domain),
```

Comment:
Did you make the changes such that I can set a setting for the public ports to be not 80, 443, but something like 8080 and 8443

Reply:
yes, but I forgot to change here to respect the value.